src/backend_model.cc

```diff
@@ -250,17 +241,71 @@ TritonModel::Create(
     }
   }
 
-  // Create and initialize the model instances for this model.
-  RETURN_IF_ERROR(TritonModelInstance::CreateInstances(
+  // Create or update the model instances for this model.
+  RETURN_IF_ERROR(TritonModelInstance::SetInstances(
       raw_local_model, backend_cmdline_config_map, host_policy_map,
-      model_config, device_blocking));
+      model_config));
+  RETURN_IF_ERROR(local_model->CommitInstances());
 
   RETURN_IF_ERROR(local_model->SetConfiguredScheduler());
```

Comment:
Swapping the scheduler without proper check for ongoing sequences in sequence batcher can cause some of the requests for allocated streams to run into first request doesn't specify START flag.

The future requests for a sequence allocated in previous scheduler should still be routed through the same scheduler. I believe the only accurate solution for scaling instances up/down should be to dynamically adjust it within the same sequence_batcher object. @GuanLuo feel free to provide your suggestions/correction here.

Reply:
Yes, but please correct me if I am wrong. I think the same issue currently exists with the `model_lifecycle` model update, because it will swap from pointing to the background model to the foreground, once the new foreground model loading is complete, without knowing anything about the scheduler. Then, the same problem can happen if the swap happens in the middle of a sequence.

I think another possible fix is to block any model update (including instance update) on the model with ongoing sequence(s) until the ongoing sequence(s) completes or times out. At the same time, any other inference requests, not related to the ongoing sequence(s), will pause until the update completes.

If the issue is fixed for model instance update now, we will introduce a behavior change for the type of updates that was previously handled by the `model_lifecycle` update logic. Do you think we might just disable model instance update on sequence model for now, and fix both the `model_lifecycle` and 'instance update' sequence model issue in another ticket?

Reply:
> Do you think we might just disable model instance update on sequence model for now, and fix both the model_lifecycle and 'instance update' sequence model issue in another ticket?

Yes. I will let @GuanLuo  comment on the model reloads with sequence batcher. As per my current reading, I too think we will run into same issue when re-loading any model with sequence batching on.